framework/src/meshgenerators/XYZDelaunayGenerator.C

```diff
@@ -0,0 +1,408 @@
+  for (auto hole_i : index_range(_hole_ptrs))
+  {
+    if (hole_i < _stitch_holes.size() && _stitch_holes[hole_i])
+    {
+      UnstructuredMesh & hole_mesh = dynamic_cast<UnstructuredMesh &>(**_hole_ptrs[hole_i]);
+      auto & hole_boundary_info = hole_mesh.get_boundary_info();
+
+      // Our algorithm here requires a serialized Mesh.  To avoid
+      // redundant serialization and deserialization (libMesh
+      // MeshedHole and stitch_meshes still also require
+      // serialization) we'll do the serialization up front.
+      MeshSerializer serial_hole(hole_mesh);
+
+      // We'll look for any sides that match between the hole mesh and
+      // the newly triangulated mesh, and apply bcids accordingly.
+      for (auto elem : hole_mesh.element_ptr_range())
+        for (auto s : make_range(elem->n_sides()))
+          if (!elem->neighbor_ptr(s))
+          {
+            auto points = sorted_point_tuple(*elem, s);
+            auto it = mesh_faces.find(points);
+
+            // I'd love to assert that we don't have any missing
+            // matches, but our holes might themselves have holes
+            if (it != mesh_faces.end())
+            {
+              auto [main_elem, main_side] = it->second;
+              hole_boundary_info.add_side(elem, s, new_hole_bcid);
+              mesh_boundary_info.add_side(main_elem, main_side, inner_bcid);
+            }
+          }
+
+      // Retrieve subdomain name map from the mesh to be stitched and insert it into the main
+      // subdomain map
+      const auto & increment_subdomain_map = hole_mesh.get_subdomain_name_map();
+      main_subdomain_map.insert(increment_subdomain_map.begin(), increment_subdomain_map.end());
+
+      std::size_t n_nodes_stitched = mesh->stitch_meshes(hole_mesh,
+                                                         inner_bcid,
+                                                         new_hole_bcid,
+                                                         TOLERANCE,
+                                                         /*clear_stitched_bcids*/ true,
+                                                         _verbose_stitching,
+                                                         use_binary_search);
+
+      if (!n_nodes_stitched)
+        mooseError("Failed to stitch hole mesh ", hole_i, " to new tetrahedralization.");
+    }
+  }
+  // Check if one SubdomainName is shared by more than one subdomain ids
+  std::set<SubdomainName> main_subdomain_map_name_list;
+  for (auto const & id_name_pair : main_subdomain_map)
+    main_subdomain_map_name_list.emplace(id_name_pair.second);
+  if (main_subdomain_map.size() != main_subdomain_map_name_list.size())
+    paramError("holes", "The hole meshes contain subdomain name maps with conflicts.");
+
+  // We're done with the hole meshes now, and MeshGenerator doesn't
+  // want them anymore either.
+  for (auto & hole_ptr : _hole_ptrs)
+    hole_ptr->reset();
+
+  mesh->prepare_for_use();
```

Comment:
I think best practice now is to set_isnt_prepare and prepare only when needed in the MGs that require it